File: Scala/src/Sources/Count/count_short.scala
package Sources.Count

import Others.value.rand_p

import scala.collection.mutable.ArrayBuffer

object count_short {
    private def count_sort_short(a: Array[Int]): Unit = {
        var l = a(0)
        for (i <- a.indices) {
            if (a(i) >= l)
                l = a(i)
        }

        val b = ArrayBuffer.fill(l + 1)(0)

        for (i <- a.indices) {
            b(a(i)) += 1
        }

        var k = 0
        for (i <- 0 until l + 1) {
            for (_ <- 0 until b(i)) {
                a(k) = i
                k += 1
            }
        }
    }

    def main(args: Array[String]): Unit = {
        val s = 1000000 // Changeable size
        val a = rand_p(s)

        val start = System.currentTimeMillis()

        count_sort_short(a)

        val end = System.currentTimeMillis() - start

        for (i <- a.indices)
            println(a(i))
        print(end + " ms")
    }
}

File: Kotlin/src/main/kotlin/Sources/Count/count_short.kt
package Sources.Count

fun count_short(a: IntArray) {
    var l = a[0]
    for (i in a.indices) {
        if (a[i] >= l)
            l = a[i]
    }

    val b: MutableList<Int> = MutableList(l+1){0}

    for (i in 0 until a.size)
        ++b[a[i]]

    var k=0
    for (i in 0 until l+1) {
        for (j in 0 until b[i])
            a[k++] = i
    }
}

fun main() {
    val a = intArrayOf(5, 4, 3, 2, 1)

    val start = System.currentTimeMillis()

    count_long(a)

    val end = System.currentTimeMillis() - start

    for (i in a.indices)
        println(a[i])
    print("$end ms")
}